ttax/autodiff_test.py

```diff
@@ -44,5 +44,22 @@ def testDoubleProjection(self):
     double_project = autodiff.project(project, vector)
     self.assertAllClose(ops.full(project), ops.full(double_project), rtol=1e-4)
 
+  def testHessianVectorProduct(self):
+      rng1, rng2, rng3 = jax.random.split(jax.random.PRNGKey(0), 3)
+      dtype = jnp.float32
+      shape = (5, 5, 5)
+      A = random_.matrix(rng1, (shape, shape), dtype=dtype)
+      AT = ops.transpose(A)
+      A_plus_AT = A + AT
+      x = random_.tensor(rng2, shape, dtype=dtype)
```

Comment:
I don't know if this works, but it isn't supposed to: the idea was that tensors and matrices don't mix. You should only multiply matrices by matrices (that can be N x 1). In the case when you do want to mix them, you should explicitely convert (something like ttax.tensor_to_vector, to be implemented).

If you strongly feel otherwise, let's discuss. If not, can you please change it to
`random_.tensor(rng2, (shape, None), dtype=dtype)`